homework/index.js

```diff
@@ -30,17 +32,161 @@
     return elem;
   }
 
-  function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
-      } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
+  function formatDateString(string) {
+    const date = new Date(string);
+    const options = {
+      weekday: 'long',
+      year: 'numeric',
+      month: 'long',
+      day: 'numeric',
+      hour: '2-digit',
+      minute: '2-digit',
+    };
+    return date.toLocaleDateString('en-GB', options);
```

Comment:
Rather than specifying exactly how the date should be formatted you can also (and in many cases, preferably) rely on the computer default locale. This is what the user has set up in the computer settings. For instance, on my computer I have defined a Dutch locale, with Dutch rules for date and time. A US user might expect a US date and time format. If you use `.toLocaleDateString()` the defaults will be applied and your code is simpler too.